### DISCUSSION TITLE
zigpy/util.py catching errors !!

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#755] [ZHA development]
I have a Listener defined here: https://github.com/pipiche38/Domoticz-Zigpy/blob/master/Classes/zp_listener.py

However if I have python3 errors in the Listener code this is catch by the try:/Except code in util.py 

In other words, if there is any coding errors in the zp_listener.py  they are caught by a try/except in util.py but no stack, nor module, nor line number are displayed. 

This make the python development using zigpy quiet difficult.

Should the zigpy/util raise an Exception instead of intercepting the error ?
https://github.com/zigpy/zigpy/blob/dev/zigpy/util.py#L43

**Comment by puddly** [comment]
Adding `exc_info=e` to the logging line should log the full traceback as a warning.

**Reply by pipiche38** [reply]
thanks

**Reply by pipiche38** [reply]
Do I need to write a PR for that ?

**Reply by puddly** [reply]
Sure. I think some of these warnings are logged by the `appdb` module so hopefully it won't make the log too verbose.
